cf-atcp-emailer/src/main/java/org/broadinstitute/ddp/cf/ATCPContactEmailer.java

```diff
@@ -0,0 +1,191 @@
+
+    public ATCPContactEmailer() {
+        String sendgridApiKey = getEnvOrThrow("SENDGRID_API_KEY");
+        sendgridFromName = getEnvOrThrow("SENDGRID_FROM_NAME");
+        sendgridFromEmail = getEnvOrThrow("SENDGRID_FROM_EMAIL");
+        sendgridToEmail = getEnvOrThrow("SENDGRID_TO_EMAIL");
+        templateId = getEnvOrThrow("EMAIL_TEMPLATE");
+        expectedBucket = getEnvOrThrow("BUCKET");
+        String gcpProject = getEnvOrThrow("GCP_PROJECT");
+
+        GoogleCredentials googleCredentials;
+        try {
+            googleCredentials = GoogleCredentials.getApplicationDefault();
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Cannot obtain GoogleCredentials", e);
+        }
+
+        storage = StorageOptions.newBuilder()
+                .setCredentials(googleCredentials)
+                .setProjectId(gcpProject)
+                .build()
+                .getService();
+
+        this.sendgrid = new SendGrid(sendgridApiKey);
+    }
+
+    private String getEnvOrThrow(String name) {
+        String value = System.getenv(name);
+        if (value == null || value.isBlank()) {
+            throw new IllegalArgumentException("Missing environment variable: " + name);
+        }
+        return value;
+    }
+
+    @Override
+    public void accept(FileScanResult event, Context context) {
+        logger.info("Received message: eventId=" + context.eventId() + " timestamp=" + context.timestamp());
+
+        String bucketName = event.getAttributes().get(ATTR_BUCKET_ID);
+        String filename = event.getAttributes().get(ATTR_OBJECT_ID);
+        String scanResult = event.getAttributes().get(ATTR_SCAN_RESULT);
+        logger.info(String.format("File scan result: bucket=%s file=%s result=%s", bucketName, filename, scanResult));
+
+        if (bucketName == null || bucketName.isBlank()) {
+            logger.severe("Event message is missing name of bucket");
+            return;
+        } else if (filename == null || filename.isBlank()) {
+            logger.severe("Event message is missing name of file");
+            return;
+        } else if (!expectedBucket.equals(bucketName)) {
+            logger.severe("Event message is not for expected bucket: " + expectedBucket);
+            return;
+        } else if (!FILENAME_PATTERN.matcher(filename).matches()) {
+            // It might be the metadata file, so skip.
+            logger.info("Filename pattern does not match, skipping");
+            return;
+        }
+
+        ScanResult result;
+        try {
+            result = ScanResult.valueOf(scanResult);
+        } catch (Exception e) {
+            throw new RuntimeException("Unexpected file scan result: " + scanResult, e);
+        }
+
+        if (result == ScanResult.INFECTED) {
+            throw new RuntimeException(String.format(
+                    "File is infected! Will not send ATCP DAR email! (bucket=%s file=%s)", bucketName, filename));
+        }
```

Comment:
This is the most significant change -- we're now inspecting the file scan result! I don't know what we want to do if we get an infected file. For now, the function will halt and yell loudly.

Reply:
I think it's fine, there's nothing we can do in this case. Maybe we should just remove the file?

Reply:
Yeah, you know what, I think we should remove the file but keep the metadata file, so we have some trace of what it is.

Reply:
Done! 2dbe19f097b841736422308d06ccf6d858527402